modules/ismp/pallet/pallet-asset-transfer/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+
+        let mut to = [0u8; 32];
+        let _temp: [u8; 20] = multi_account.evm_account.clone().into();
+        to.iter_mut().enumerate().for_each(|(index, item)| {
+            if index < 20 {
+                *item = _temp[index];
+            }
+        });
+        let from: [u8; 32] = multi_account.substrate_account.clone().into();
+        let asset_id = T::DotAssetId::get().0.into();
+        let body = Body {
+            amount: {
+                let amount: u128 = amount.into();
+                let mut bytes = [0u8; 32];
+                U256::from(amount).to_big_endian(&mut bytes);
+                alloy_primitives::U256::from_be_bytes(bytes)
+            },
+            asset_id,
+            redeem: false,
+            from: from.into(),
+            to: to.into(),
+        };
+
+        let dispatch_post = DispatchPost {
+            dest: multi_account.dest_state_machine,
+            from: T::TokenGateWay::get().0.to_vec(),
+            to: T::TokenGateWay::get().0.to_vec(),
+            // 1 hour timeout
+            timeout_timestamp: 60 * 60,
+            data: alloy_rlp::encode(body),
+        };
+
+        dispatcher
+            .dispatch_request(
+                DispatchRequest::Post(dispatch_post),
+                multi_account.substrate_account.clone(),
+                Default::default(),
+            )
+            .map_err(|_| Error::<T>::DispatchPostError)?;
+
+        Self::deposit_event(Event::<T>::TransferInitiated {
+            from: multi_account.substrate_account,
+            to: multi_account.evm_account,
+            dest: multi_account.dest_state_machine,
+            amount,
+        });
+
+        Ok(())
+    }
+}
+
+#[derive(RlpDecodable, RlpEncodable, Debug, Clone)]
+pub struct Body {
+    // amount to be sent
+    pub amount: alloy_primitives::U256,
+    // The token identifier
+    pub asset_id: alloy_primitives::B256,
+    // flag to redeem the erc20 asset on the destination
+    pub redeem: bool,
+    // sender address
+    pub from: alloy_primitives::B256,
+    // recipient address
+    pub to: alloy_primitives::B256,
+}
+
+#[derive(Clone)]
+pub struct Module<T>(PhantomData<T>);
+
+impl<T: Config> Default for Module<T> {
+    fn default() -> Self {
+        Self(PhantomData)
+    }
+}
+
+impl<T: Config> IsmpModule for Module<T>
+where
+    <T::Assets as fungibles::Inspect<T::AccountId>>::Balance: From<u128>,
+    u128: From<<T::Assets as fungibles::Inspect<T::AccountId>>::Balance>,
+    <T::Assets as fungibles::Inspect<T::AccountId>>::AssetId: From<MultiLocation>,
+    T::AccountId: Into<[u8; 32]> + From<[u8; 32]>,
+    T::EvmAccountId: Into<[u8; 20]>,
+{
+    fn on_accept(&self, post: ismp::router::Post) -> Result<(), ismp::error::Error> {
+        let request = Request::Post(post.clone());
+        // Check that destination module is equal to the known module id
+        ensure!(
+            request.destination_module() == T::TokenGateWay::get().0.to_vec(),
```

Comment:
meh, this check isn't neccessary i believe the router already does this, i think we instead want to check that the request originates from the `TokenGateway` contract address